WMCO 10.18.0/4.18 release notes (#87287)
openshift/openshift-docs: windows_containers/wmco_rn/windows-containers-release-notes-10-18-x.adoc

:_mod-docs-content-type: ASSEMBLY
[id="windows-containers-release-notes-10-18-x"]
= {productwinc} release notes
include::_attributes/common-attributes.adoc[]
:context: windows-containers-release-notes

toc::[]

[id="wmco-10-18-0"]
== Release notes for Red Hat Windows Machine Config Operator 10.18.0

> 

Author: Done

This release of the Windows Machine Config Operator (WMCO) provides bug fixes for running Windows compute nodes in an {product-title} cluster. The components of the WMCO 10.18.0 were released in link:https://access.redhat.com/errata/RHSA-2024:TBD[TBD].

[id="wmco-10-18-0-new-features"]
=== New features and improvements

[id="wmco-10-18-0-new-features-hpa"]
==== Horizontal Pod autoscaling is available for Windows workloads.
You can now use a horizontal pod autoscaler (HPA) to scale your Windows pods based on CPU and/or memory resource metrics. For more information on using an HPA, see xref:../../nodes/pods/nodes-pods-autoscaling.adoc#nodes-pods-autoscaling[Automatically scaling pods with the horizontal pod autoscaler].

[id="wmco-10-18-0-new-features-eus"]
==== Control Plane Only updates are now available for Windows nodes

> You can now use *Control Plane Only* updates, previously known as *EUS-to-EUS* updates, to update your Windows nodes between {product-title} EUS version. For more information, see windows_containers/windows-node-upgrades.adoc#wmco-upgrades-eus_windows-node-upgrades[Windows Machine Config Operator Control Plane Only upgrade].

Author: @cbippley Apologies! I didn't hook up the links as the new modules aren't present in the docs yet.

[id="wmco-10-18-0-new-features"]
=== New features and improvements

[id="wmco-10-18-0-new-features-hpa"]
==== Horizontal Pod autoscaling is available for Windows workloads
You can now use a horizontal pod autoscaler (HPA) to scale your Windows pods based on CPU and/or memory resource metrics. For more information on using an HPA, see xref:../../nodes/pods/nodes-pods-autoscaling.adoc#nodes-pods-autoscaling[Automatically scaling pods with the horizontal pod autoscaler].

[id="wmco-10-18-0-new-features-eus"]
==== Control Plane Only updates are now available for Windows nodes
You can now use *Control Plane Only* updates, previously known as *EUS-to-EUS* updates, to update your Windows nodes between {product-title} EUS versions. For more information, see xref:../../windows_containers/windows-node-upgrades.adoc#wmco-upgrades-eus_windows-node-upgrades[Windows Machine Config Operator Control Plane Only upgrade].

[id="wmco-10-18-0-new-features-metrics"]
==== WMCO metrics endpoint is now HTTPS

> The WMCO metrics endpoint is now exposed over HTTPS. Previously, WMCO pod metrics were available over HTTP. This change ensures that the WMCO metrics endpoint is secure.

Reviewer: I worry sometimes about strong absolute statements, esp wrt security. I don't know enough about this, but would it make sense to say:
```suggestion
The WMCO metrics endpoint is now exposed over HTTPS. Previously, WMCO pod metrics were available over HTTP. This change improves the security posture of the WMCO metrics endpoint.
```

// https://issues.redhat.com/browse/WINC-1269
// https://issues.redhat.com/browse/WINC-588

[id="wmco-10-18-0-new-features-debuglogging"]
==== WMCO now defaults to info-level logging

> The Windows Machine Config Operator (WMCO) is now configured to use the `info` log level by default. You can change the log level to `debug` by editing the WMCO subscription object. For more information, see xref:../../windows_containers/enabling-windows-container-workloads.adoc#wmco-configure-debug-logging_enabling-windows-container-workloads[Configuring debug-level logging for the Windows Machine Config Operator].

Reviewer: Already defined above
```suggestion
The WMCO is now configured to use the `info` log level by default. You can change the log level to `debug` by editing the WMCO subscription object. For more information, see xref:../../windows_containers/enabling-windows-container-workloads.adoc#wmco-configure-debug-logging_enabling-windows-container-workloads[Configuring debug-level logging for the Windows Machine Config Operator].
```

[id="wmco-10-18-0-new-features-kubernetes"]
==== Kubernetes upgrade
The WMCO now uses Kubernetes 1.31.

[id="wmco-10-18-0-bug-fixes"]
=== Bug fixes
* Previously, if you installed the WMCO into the the default `openshift-windows-machine-config-operator` namespace and removed the WMCO, subsequently installing the WMCO into a namespace other than the default would fail. This was because the role-based access control (RBAC) for the Windows Instance Config Daemon (WICD) was always associated with the default namespace. With this fix, when installing the WMCO to a non-default namespace, the RBAC for the WICD is configured correctly. (link:https://issues.redhat.com/browse/OCPBUGS-46473[*OCPBUGS-46473*])

> * Previously, the {product-title} documentation lacked information about WMCO support for the installer-provisioned infrastructure (IPI) and the user-provisioned infrastructure (UPI) installation methods. With this fix, the documentation now reports that the IPI method is the preferred installation method and can be used with all platforms. The UPI method is limited to specific use cases. For more information, see xref:../../windows_containers/wmco_rn/windows-containers-release-notes-10-18-x-prereqs.adoc#wmco-prerequisites-supported-install-10.18.0_windows-containers-release-notes-10-18-x-prereqs[WMCO supported installation method]. (link:https://issues.redhat.com/browse/OCPBUGS-18898[*OCPBUGS-18898*])

Reviewer: I am fairly certain that IPI and UPI are not really approved abbreviations. There are 6 hits for (IPI) in the repo, so it's not something that's used a ton in the install books, where I would expect to see it if it were approved. I would recommend avoiding these abbreviations unless an install writer can approve the use.
